builtin/k8s/platform.go

```diff
@@ -125,6 +128,164 @@ func (p *Platform) getClientset() (*clientsetInfo, error) {
 	}, nil
 }
 
+func (p *Platform) resourceDeploymentStatus(
+	ctx context.Context,
+	log hclog.Logger,
+	sg terminal.StepGroup,
+	deploymentState *Resource_Deployment,
+	clientset *clientsetInfo,
+	sr *resource.StatusResponse,
+) error {
+	s := sg.Add("Checking status of the kubernetes deployment resource...")
+	defer s.Abort()
+
+	namespace := p.config.Namespace
+	if namespace == "" {
+		namespace = clientset.Namespace
+	}
+
+	// Get deployment status
+
+	deploymentResource := sdk.StatusReport_Resource{
+		Platform:            platformName,
+		Type:                "deployment",
+		CategoryDisplayHint: sdk.ResourceCategoryDisplayHint_INSTANCE_MANAGER,
+	}
+	sr.Resources = append(sr.Resources, &deploymentResource)
+
+	deployResp, err := clientset.Clientset.AppsV1().Deployments(namespace).Get(ctx, deploymentState.Name, metav1.GetOptions{})
+	if err != nil {
+		if errors.IsNotFound(err) {
+			deploymentResource.Name = deploymentState.Name
+			deploymentResource.Health = sdk.StatusReport_MISSING
+			deploymentResource.HealthMessage = sdk.StatusReport_MISSING.String()
+		} else {
+			return status.Errorf(codes.FailedPrecondition, "error getting kubernetes deployment %s: %s", deploymentState.Name, err)
+		}
+	} else if deployResp == nil {
+		return status.Errorf(codes.FailedPrecondition, "kubernetes deployment response cannot be nil")
+	} else {
```

Comment:
This is more of a code nit but I think we can break this out from the `else` since the other two cases are returning for nil/err checking.

Reply:
If the deployment is missing, we won't return early but we still don't want to execute the big chunk that depends on a populated `deployResp`